### PR TITLE
Create the documentation with typedoc

### PR DESCRIPTION
**Pull request by MichaelKim20**
SDK documents are automatically generated when code is pushed.

Related to #22

### CODE REVIEW

**Comment by TrustHenry** [comment]
I'll test it.

**Comment by AndrejMitrovic** [comment]
@Geod24 Michael says there's errors in the documentation build here, but I don't understand what's wrong. I thought maybe `github.actors` might be a secret that wasn't set for the repo, but that's not the problem.

**Comment by Geod24** [comment]
No `github.actor` is just the user name of the person triggering the action.
It looks like the program is trying to push to github using the HTTP endpoint, not the git+ssh endpoint, and then trying to read the username/password, but there's no tty (it's not interactive) and it's the message we're seeing.

**Comment by MichaelKim20** [comment]
@Geod24
I did the following for the test.
1. I did PR in my personal repository. ( branch name is `tsdoc-test`)   
    It has scripts and Github actions that produce documents.
    https://github.com/MichaelKim20/boa-sdk-ts/pull/1
2. I merged PR into my personal repository.
    When merging, documents were created and pushed to the branch `gh-pages`.
    https://github.com/MichaelKim20/boa-sdk-ts/runs/939383218
3. I checked the generated page. However, there was no access to subfolders.
     https://michaelkim20.github.io/boa-sdk-ts/
     https://michaelkim20.github.io/boa-sdk-ts/modules/_modules_data_preimage_.html

**Comment by Geod24** [comment]
Well it should have access to subfolders. I don't know why they don't show up, it's weird, but definitely not what should happen. Could it be due to the extra underscores ? I would check with the project that generates the doc first, and Github support second.

**Comment by MichaelKim20** [comment]
I changed the options for typedoc and solved it so that server folders can be made in 'github pages'.
I checked that the document is being generated normally.
https://michaelkim20.github.io/boa-sdk-ts/
When this PR is merged, The document will be created in the upstream.

**Comment by MichaelKim20** [comment]
The github action script that produces Agora's gh-pages helped me a lot in working on this.

File: .gitignore [hunk 1]
```diff
@@ -92,3 +92,6 @@ dist
 .idea
 
 lib
+
+# SKD document
```
Comment (Geod24) [inline .gitignore:96]:
```suggestion
# Documentation build folder
```

Note that "document" and "documentation" are two very different words and cannot be used interchangeably.

Comment (TrustHenry) [reply .gitignore]: I improved the wording.

**Comment by Geod24** [comment]
Great work!

**Pull request merged**